src/mbgl/renderer/layers/render_fill_layer.cpp

```diff
@@ -93,29 +93,23 @@ void RenderFillLayer::render(PaintParameters& parameters, RenderSource*) {
                 );
             };
 
-            if (evaluated.get<FillAntialias>() && !unevaluated.get<FillOutlineColor>().isUndefined() && parameters.pass == RenderPass::Translucent) {
-                draw(2,
-                     parameters.programs.fillOutline,
-                     gl::Lines { 2.0f },
-                     *bucket.lineIndexBuffer,
-                     bucket.lineSegments);
-            }
-
             // Only draw the fill when it's opaque and we're drawing opaque fragments,
             // or when it's translucent and we're drawing translucent fragments.
             if ((evaluated.get<FillColor>().constantOr(Color()).a >= 1.0f
               && evaluated.get<FillOpacity>().constantOr(0) >= 1.0f) == (parameters.pass == RenderPass::Opaque)) {
-                draw(1,
-                     parameters.programs.fill,
+                draw(parameters.programs.fill,
                      gl::Triangles(),
+                     parameters.depthModeForSublayer(1, gl::DepthMode::ReadWrite),
                      *bucket.triangleIndexBuffer,
                      bucket.triangleSegments);
             }
 
-            if (evaluated.get<FillAntialias>() && unevaluated.get<FillOutlineColor>().isUndefined() && parameters.pass == RenderPass::Translucent) {
-                draw(2,
-                     parameters.programs.fillOutline,
-                     gl::Lines { 2.0f },
+            if (evaluated.get<FillAntialias>() && parameters.pass == RenderPass::Translucent) {
+                draw(parameters.programs.fillOutline,
+                     gl::Lines{ 2.0f },
+                     parameters.depthModeForSublayer(
+                         unevaluated.get<FillOutlineColor>().isUndefined() ? 2 : 0,
```

Comment:
You're right. However, upon further inspection, we're using different formulas to compute the depth mode:

JS:
```js
const farDepth = 1 - ((1 + this.currentLayer) * this.numSublayers + n) * this.depthEpsilon;
const nearDepth = farDepth - 1 + this.depthRange;
```

Native:
```cpp
const float nearDepth = ((1 + currentLayer) * numSublayers + n) * depthEpsilon;
const float farDepth = nearDepth + depthRangeSize;
```

I'll ticket this as a separate change that we'll have to make.